src/LibChatMessage.lua

```diff
@@ -1,549 +1,622 @@
+    return osdate(lib.settings.timePrefixFormat, timeStamp)
+end
+
+local function GetTimeStampForEvent()
+    if(lib.nextEventTimeStamp) then
+        return lib.nextEventTimeStamp, true
+    end
+    return GetTimeStamp(), false
+end
+
+local function StoreChatEvent(timeStamp, type, ...)
+    if(not lib.chatHistoryActive) then return end
+    local chatHistory = lib.chatHistory
+    chatHistory[#chatHistory + 1] = {timeStamp, type, ...}
+    if(#chatHistory > MAX_HISTORY_LENGTH) then
+        local newHistory = {}
+        for i = #chatHistory - TRIMMED_HISTORY_LENGTH, #chatHistory do
+            newHistory[#newHistory + 1] = chatHistory[i]
+        end
+        chatHistory = newHistory
+    end
+end
+
+local function ApplyTimeAndTagPrefix(formattedEventText, targetChannel, fromDisplayName, rawMessageText, timeStamp)
+    if(formattedEventText) then
+        if(lib.settings and lib.settings.tagPrefixMode ~= TAG_PREFIX_OFF) then
+            formattedEventText = MESSAGE_TEMPLATE:format(SYSTEM_TAG, formattedEventText)
+        end
+        if(lib.settings and lib.settings.timePrefixEnabled) then
+            formattedEventText = MESSAGE_TEMPLATE:format(GetFormattedTime(timeStamp), formattedEventText)
+        end
+    end
+    return formattedEventText, targetChannel, fromDisplayName, rawMessageText
+end
+
+-- chat system hooks
+local messageFormatters = CHAT_ROUTER:GetRegisteredMessageFormatters()
+local newFormatter = {}
+do
+    local redirect = {}
+    setmetatable(redirect, newFormatter)
+    redirect.__index = function(_, key)
+        -- Called, if newFormatter has no entry: use original.
+        return messageFormatters[key]
+    end
+    SecurePostHook(CHAT_ROUTER, "FormatAndAddChatMessage", function(self)
+        -- Restore Formatters for addons
+        self.registeredMessageFormatters = messageFormatters
+    end)
+    local orgOnChatEvent = CHAT_ROUTER.FormatAndAddChatMessage
+    function CHAT_ROUTER.FormatAndAddChatMessage(self, ...)
+        if IsChatSystemAvailableForCurrentPlatform() then
+            -- Replace Formatters for ZOS
+            self.registeredMessageFormatters = newFormatter
+        end
+        return orgOnChatEvent(self, ...)
+    end
+end
+
+local function dummyPreHook(...) return ... end
+local function PostHookFormatter(eventType, postHook, preHook)
+    preHook = preHook or dummyPreHook
+    newFormatter[eventType] = function(...)
+        local timeStamp, isRestoring = GetTimeStampForEvent()
+        if(not isRestoring) then
+            StoreChatEvent(timeStamp, eventType, ...)
+        end
+        local originalFormatter = messageFormatters[eventType]
+        local formattedEventText, targetChannel, fromDisplayName, rawMessageText = originalFormatter(preHook(...))
+        return postHook(formattedEventText, targetChannel, fromDisplayName, rawMessageText, timeStamp)
+    end
+end
+
+----------------------------------------------
+
+UNKNOWN_LINK_TYPE = "unknown"

```

Comment:
Make it a local and if it has to be used by other addons, expose it via the lib table as lib.UNKNOWN_LINK_TYPE.